ios-target/View/Profile/ProfileViewController.swift

```diff
@@ -10,9 +10,64 @@ import UIKit
 
 class ProfileViewController: UIViewController {
   
+  @IBOutlet weak var usernameTextField: UITextFieldPadding!
+  @IBOutlet weak var emailTextField: UITextFieldPadding!
+  @IBOutlet weak var firstNameTextField: UITextFieldPadding!
+  @IBOutlet weak var lastNameTextField: UITextFieldPadding!
+  @IBOutlet weak var saveProfileButton: UIButton!
+  
   var viewModel = ProfileViewModel()
   
   override func viewDidLoad() {
     super.viewDidLoad()
+    viewModel.delegate = self
+    saveProfileButton.enable(false)
+    viewModel.loadUserProfile()
```

Comment:
why do you need to load user profile here? shouldn't this information be in the session manager already?

Reply:
just checked and there is a currentUser in the dataManager 😅. There is also a url in the api for this so i figured it was meant to be implemented that way 🤷🏽‍♀️

Reply:
I think it is a good idea to fetch the updated information. It should be a background process though, without blocking anything nor alerting the user.